src/main/java/org/computerspecsviewer/displaytypes/customduration/CustomDuration.java

```diff
@@ -3,25 +3,77 @@
 import java.time.Duration;
 
 public class CustomDuration {
+    private static final String[] durationInfoStrings = new String[]{"days", "hours", "minutes", "seconds"};
+
     public Duration duration;
+    public DurationLimit durationLimit;
+    public boolean isNegative;
 
     public CustomDuration(long seconds) {
-        duration = Duration.ofSeconds(seconds);
+        this(seconds, DurationLimit.UP_TO_DAYS);
+    }
+
+    public CustomDuration(long seconds, DurationLimit durationLimit) {
+        if(seconds >= 0) {
+            this.duration = Duration.ofSeconds(seconds);
+            isNegative = false;
+        } else {
+            isNegative = true;
+        }
+
+        this.durationLimit = durationLimit;
     }
 
     @Override
     public String toString() {
-        long dayAmount = duration.toDays();
+        if(isNegative) {
+            return "Cannot be calculated";
+        }
+
+        Duration currentDuration = duration;
+        long dayAmount = -1;
+        long hourAmount = -1;
+        long minuteAmount = -1;
+
+        if(durationLimit.compareTo(DurationLimit.UP_TO_DAYS) == 0) {
```

Comment:
Is there any better way to implement this other than using a bunch of if statements?